plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java

```diff
@@ -464,6 +463,38 @@ public Object convert(Object fromObject) {
     }
   }
 
+  private static class ProjectSelectionValidator extends FixedMultiValidator {
+
+    private final IViewerObservableValue projectInput;
+    private final IViewerObservableValue projectSelection;
+    private final boolean requireValues;
+
+    private ProjectSelectionValidator(IViewerObservableValue projectInput,
+                                      IViewerObservableValue projectSelection,
+                                      boolean requireValues) {
+      this.projectInput = projectInput;
+      this.projectSelection = projectSelection;
+      this.requireValues = requireValues;
+    }
+
+    @Override
+    protected IStatus validate() {
+      // this access is recorded and ensures that changes are tracked
+      Collection<?> projects = (Collection<?>) projectInput.getValue();
+      // this access is recorded and ensures that changes are tracked, don't move it inside the if
+      Object selectedProject = projectSelection.getValue();
+      if (projects.isEmpty()) {
+        return ValidationStatus.info(Messages.getString("projectselector.no.projects"));
```

Comment:
I see one problem when returning INFO. The _Deploy_ button is enabled.

![deploy to app engine standard _039](https://cloud.githubusercontent.com/assets/10523105/23720468/c0eced98-040c-11e7-87e0-e00638272dcb.png)

However, if we change to return ERROR, it should go into `if (requirevalues)`, so that users can save settings in the preference page.